### PR TITLE
fix: actually catch the error code from MoveFileEx

### PR DESCRIPTION
**Pull request by eric-wieser**
A user on Zulip [reported seeing an error code of "no error"](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.60lake.20update.60.20broken.20on.20Windows.20.28.3F.29/near/429134334) here.

### CODE REVIEW

File: src/runtime/io.cpp [hunk 1]
```diff
@@ -788,6 +788,9 @@ extern "C" LEAN_EXPORT obj_res lean_io_rename(b_obj_arg from, b_obj_arg to, lean
     // so we have to call the underlying windows API directly to get behavior consistent
     // with the unix-like OSs
     bool ok = MoveFileEx(string_cstr(from), string_cstr(to), MOVEFILE_REPLACE_EXISTING) != 0;
+    if (!ok) {
+        errno = GetLastError();
```
Comment (eric-wieser) [inline src/runtime/io.cpp:792]: Actually, I think some conversion is probably needed here...

**Comment by tydeu** [comment]
@eric-wieser  Thanks for finding this! Unfortunately, as you surmised, `GetLastErrror` does not directly map to `errno`, so some additional conversion is necessary to properly fix this error.

File: src/runtime/io.cpp (hunk 1 above)
Comment (Kha) [reply src/runtime/io.cpp:792]: See other uses of `GetLastError`, we usually use the string overload of `io_result_mk_error` for lack of a conversion function

Comment (eric-wieser) [reply src/runtime/io.cpp:792]: In the longer run perhaps it's worth copying [`errmap` from CPython](https://github.com/python/cpython/blob/main/PC/errmap.h#L2)?

**Review by Kha — APPROVED** [review]

**Pull request merged**